specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimversionsets.json

```diff
@@ -0,0 +1,368 @@
+                    }
+                },
+                "parameters": [
+                    {
+                        "name": "parameters",
+                        "in": "body",
+                        "required": true,
+                        "schema": {
+                            "$ref": "#/definitions/ApiVersionSetContract"
+                        },
+                        "description": "Update parameters."
+                    },
+                    {
+                        "name": "If-Match",
+                        "in": "header",
+                        "required": true,
+                        "description": "The entity state (Etag) version of the user to update. A value of \"*\" can be used for If-Match to unconditionally apply the operation.",
+                        "type": "string"
+                    }
+                ],
+                "responses": {
+                    "204": {
+                        "description": "The Api Version Set details were successfully updated."
+                    },
+                    "default": {
+                        "description": "*** Error Responses: ***\n\n * 4XX Error response giving details why the operation failed.",
+                        "schema": {
+                            "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+                        }
+                    }
+                }
+            },
+            "delete": {
+                "tags": [
+                    "ApiVersionSets"
+                ],
+                "operationId": "ApiVersionSet_Delete",
+                "description": "Deletes specific Api Version Set.",
+                "x-ms-examples": {
+                    "ApiManagementDeleteApiVersionSet": {
+                        "$ref": "./examples/ApiManagementDeleteApiVersionSet.json"
+                    }
+                },
+                "parameters": [
+                    {
+                        "name": "If-Match",
+                        "in": "header",
+                        "required": true,
+                        "description": "The entity state (Etag) version of the Api Version Set to delete. A value of \"*\" can be used for If-Match to unconditionally apply the operation.",
+                        "type": "string"
+                    }
+                ],
+                "responses": {
+                    "204": {
+                        "description": "The ApiVersion Set details were successfully deleted."
+                    },
+                    "default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+                        }
+                    }
+                }
+            }
+        }
+    },
+    "definitions": {
+        "ApiVersionSetContractProperties": {
```

Comment:
dont we have any required property in the contract?

If yes, then we might need separate Create and Update contract.

otherwise this is fine.